crypto/srp/srp_vfy.c

```diff
@@ -24,26 +25,146 @@
 # define SRP_RANDOM_SALT_LEN 20
 # define MAX_LEN 2500
 
+/*
+ * Note that SRP uses its own variant of base 64 encoding. A different base64
+ * alphabet is used and no padding '=' characters are added. Instead we pad to
+ * the front with 0 bytes and subsequently strip off leading encoded padding.
+ * This variant is used for compatibility with other SRP implementations -
+ * notably libsrp, but also others. It is also required for backwards
+ * compatibility in order to load verifier files from other OpenSSL versions.
+ */
+
 /*
  * Convert a base64 string into raw byte array representation.
+ * Returns the length of the decoded data, or -1 on error.
  */
 static int t_fromb64(unsigned char *a, size_t alen, const char *src)
 {
-    size_t size = strlen(src);
+    EVP_ENCODE_CTX *ctx;
+    int outl = 0, outl2 = 0;
+    size_t size, padsize;
+    const unsigned char *pad = (unsigned char *)"00";
+
+    while (*src == ' ' || *src == '\t' || *src == '\n')
+        ++src;
+    size = strlen(src);
+    padsize = 4 - (size & 3);
+    padsize &= 3;
 
     /* Four bytes in src become three bytes output. */
     if (size > INT_MAX || (size / 4) * 3 > alen)
```

Comment:
Hmmm, one can actually wonder if (size/4)*3 is actually accurate. After all we'll pre-pad the string, so we do need space for ((size + padsize) / 4) * 3.

Reply:
Fixed.